src/backups.py

```diff
@@ -516,6 +519,8 @@ def _on_s3_credential_changed(self, event: CredentialsChangedEvent):
         if not self.charm.is_primary:
             return
 
+        self.charm.app_peer_data.pop("require-change-bucket-after-restore", None)
+
```

Comment:
I've added check for WAL files parity - for typical usage it works. But for situations when leader is changed there is a problem, that WAL files are unsynchronized (DB will have new .partial WAL file while S3 not). This will lead to S3 stanza will not be available to use if leader has been forcefully changed. I'm not sure if this leading to the distinct leadership issue (such like RAFT library), or how to handle this problem now.
@taurus-forever, do you have any thoughts on this?

Reply:
Replied here: https://github.com/canonical/postgresql-operator/pull/391#issuecomment-2089926343

Reply:
Closing as moved to distinct issue.